library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -311,7 +356,107 @@ public boolean subscribe(Activity activity, String productId)
 		return purchase(activity, productId, Constants.PRODUCT_TYPE_SUBSCRIPTION, null);
 	}
 
-	public boolean purchase(Activity activity, String productId, String developerPayload)
+    /**
+     *
+     * @return List of skus for past one-time items
+     */
+    public List<String> listPastHistoryProducts() {
+        return cachedProductsPast.getContents();
+    }
+
+    /**
+     *
+     * @return List of skus for past subscriptions
+     */
+    public List<String> listPastHistorySubscriptions() {
+        return cachedSubscriptionsPast.getContents();
+    }
+
+    /**
+     * Returns the most recent purchases made by the user for each SKU,
+     * even if that purchase is expired, canceled, or consumed.
+     * Note: The getPurchaseHistory() method has higher overhead than getPurchases(),
+     * because it requires a call to the Google Play server. You should use getPurchases()
+     * if you do not actually need the user's purchase history.
+     * Also is not called in onServiceConnected, but only by user's request.
```

Comment:
You shouldn't say "is not called in onServiceConnected". That leaks implementation details that clients of this code don't need to know. They just need to know this doesn't happen automatically when you instantiate a billing processor.